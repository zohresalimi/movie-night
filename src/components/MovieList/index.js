import React from "react";

export default function MovieList({ moviesList }) {
  return <div></div>;
}
