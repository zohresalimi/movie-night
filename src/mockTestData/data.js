import React, { useReducer } from "react";
import AppContext from "../store/context";
import reducer from "../store/reducer";

export const getTestState = () => {
  return {
    activePage: 1,
    movies: {
      1: {
        page: 1,
        total_pages: 2,
        total_results: 2,
        results: [
          {
            id: 10523,
            adult: false,
            poster_path: "/nJm8UF4zbUYEoku9seEtz1PTJGS.jpg",
          },
          {
            id: 50647,
            adult: false,
            poster_path: "/PlzkOA19mxEit7tV3AUnBYZ0io.jpg",
          },
        ],
      },
    },
    watchLaterList: {
      10523: {
        id: 10523,
        adult: false,
        poster_path: "/nJm8UF4zbUYEoku9seEtz1PTJGS.jpg",
      },
    },
    favoriteList: {
      10523: {
        id: 10523,
        adult: false,
        poster_path: "/nJm8UF4zbUYEoku9seEtz1PTJGS.jpg",
      },
    },
    apiConfig: {
      secure_base_url: "https://image.tmdb.org/t/p/",
      poster_sizes: ["w92", "w154", "w185", "w342", "w500", "w780"],
    },
  };
};

export const WithProvider = (props) => {
  const [state, dispatch] = useReducer(
    reducer,
    props.defaultValue || getTestState()
  );

  return (
    <AppContext.Provider
      value={{
        state,
        dispatch: props.mockDispatch || dispatch,
      }}
    >
      {props.children}
    </AppContext.Provider>
  );
};
