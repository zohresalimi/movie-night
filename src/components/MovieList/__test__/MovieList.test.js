import React from "react";
import { render, act } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import { getTestState, WithProvider } from "../../../mockTestData/data";
import MovieList from "../";

async function renderWrapper(props) {
  let component;

  act(() => {
    component = render(
      <WithProvider {...props}>
        <MovieList />
      </WithProvider>
    );
  });

  return component;
}

describe("MovieList Component Testing", () => {
  test("take snapshot", async () => {
    const { container, getByTestId } = await renderWrapper();
    expect(getByTestId("watch-list-wrapper")).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test("should match snapshot if result is empty", async () => {
    const testState = getTestState();
    const { container } = await renderWrapper({
      defaultValue: {
        ...testState,
        movies: {},
      },
    });

    expect(container).toMatchSnapshot();
  });
});
