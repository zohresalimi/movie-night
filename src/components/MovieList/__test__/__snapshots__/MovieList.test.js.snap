// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieList Component Testing should match snapshot if result is empty 1`] = `
<div>
  <div>
    <div
      class="ui grid"
      data-testid="watch-list-wrapper"
    />
  </div>
</div>
`;

exports[`MovieList Component Testing take snapshot 1`] = `
<div>
  <div>
    <div
      class="ui grid"
      data-testid="watch-list-wrapper"
    >
      <div
        class="row"
      >
        <div
          class="four wide computer sixteen wide mobile eight wide tablet column"
        >
          <div
            class="card-wrapper"
          >
            <div
              class="ui card"
            >
              <div
                class="small image dimmable"
              >
                <div
                  class="ui dimmer"
                >
                  <div
                    class="content"
                  >
                    <div>
                      <button
                        data-testid="play-btn"
                      >
                        <a
                          class="sc-bdfBwQ gaGgLj"
                        >
                          <i
                            aria-hidden="true"
                            class="play mini icon"
                          />
                        </a>
                      </button>
                      <div
                        class="tools-wrapper"
                      >
                        <div
                          class="ui small basic vertical buttons"
                        >
                          <button
                            class="ui yellow basic icon button"
                          >
                            <i
                              aria-hidden="true"
                              class="star icon"
                            />
                          </button>
                          <button
                            class="ui yellow basic icon button"
                          >
                            <i
                              aria-hidden="true"
                              class="clock icon"
                            />
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <img
                  src="https://image.tmdb.org/t/p/w500//nJm8UF4zbUYEoku9seEtz1PTJGS.jpg"
                />
              </div>
              <div
                class="content"
              >
                <div
                  class="header"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="four wide computer sixteen wide mobile eight wide tablet column"
        >
          <div
            class="card-wrapper"
          >
            <div
              class="ui card"
            >
              <div
                class="small image dimmable"
              >
                <div
                  class="ui dimmer"
                >
                  <div
                    class="content"
                  >
                    <div>
                      <button
                        data-testid="play-btn"
                      >
                        <a
                          class="sc-bdfBwQ gaGgLj"
                        >
                          <i
                            aria-hidden="true"
                            class="play mini icon"
                          />
                        </a>
                      </button>
                      <div
                        class="tools-wrapper"
                      >
                        <div
                          class="ui small basic vertical buttons"
                        >
                          <button
                            class="ui grey basic icon button"
                          >
                            <i
                              aria-hidden="true"
                              class="star icon"
                            />
                          </button>
                          <button
                            class="ui grey basic icon button"
                          >
                            <i
                              aria-hidden="true"
                              class="clock icon"
                            />
                          </button>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <img
                  src="https://image.tmdb.org/t/p/w500//PlzkOA19mxEit7tV3AUnBYZ0io.jpg"
                />
              </div>
              <div
                class="content"
              >
                <div
                  class="header"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
